siteshield/client.go

```diff
@@ -0,0 +1,164 @@
+package siteshield
+
+import (
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/corbaltcode/go-akamai"
+	"github.com/corbaltcode/go-akamai/internal/request"
+	"inet.af/netaddr"
+)
+
+const basePath = "/siteshield/v1/"
+
+// A Client allows access to the Akamai Site Shield API.
+type Client struct {
+	Credentials akamai.Credentials
+}
+
+// GetMaps returns all maps that belong to the client's account.
+func (c *Client) GetMaps() ([]Map, error) {
+	var resp mapsResp
+
+	err := request.DoJSON(c.Credentials, http.MethodGet, basePath+"maps", nil, &resp)
+	if err != nil {
+		return nil, err
+	}
+
+	maps := make([]Map, len(resp.SiteShieldMaps))
+
+	for i, mapResp := range resp.SiteShieldMaps {
+		m, err := newMapFromResp(mapResp)
+		if err != nil {
+			return nil, err
+		}
+		maps[i] = m
+	}
+
+	return maps, nil
+}
+
+// GetMap returns information about a map by ID.
+func (c *Client) GetMap(id int) (Map, error) {
+	var resp mapResp
+
+	err := request.DoJSON(c.Credentials, http.MethodGet, fmt.Sprintf("%smaps/%d", basePath, id), nil, &resp)
+	if err != nil {
+		return Map{}, err
+	}
+
+	m, err := newMapFromResp(resp)
+	if err != nil {
+		return Map{}, err
+	}
+
+	return m, nil
+}
+
+type Map struct {
+	AcknowledgeRequiredBy time.Time
+	Acknowledged          bool
+	AcknowledgedBy        string
+	Alias                 string
+	Contacts              []string
+	CurrentCIDRs          []netaddr.IPPrefix
+	ID                    int
+	IsShared              bool
+	LatestTicketID        int
+	ProposedCIDRs         []netaddr.IPPrefix
+	RuleName              string
+	Service               Service
+	SureRouteName         string
+	Type                  string
+}
+
+func newMapFromResp(r mapResp) (Map, error) {
+	var m Map
+	// m.AcknowledgeRequiredBy = time.UnixMilli(r.AcknowledgeRequiredBy)
```

Comment:
testing artifact?

Reply:
Eek, will remove. (Bug artifact! Or actually, bad Akamai documentation artifact. They said the time was in milliseconds since epoch but it's actually an RFC 3999 string.)